man/man8/dorfs.8
.\" @(#)dorfs.8 1.1 92/07/30 SMI;
.TH DORFS 8 "30 September 1988"
.SH NAME
dorfs \- initialize, start and stop RFS automatically
.SH SYNOPSIS
.B dorfs
.B init
.I domain\ \ netspec
.RI [ address ]
.br
.B dorfs
.BR start
[
.B \-v
]
.br
.B dorfs
.B stop
.SH AVAILABILITY
.LP
This program is available with the
.I \s-1RFS\s0
software installation option.
Refer to
.TX INSTALL
for information on how to install optional software.
.SH DESCRIPTION
.IX "dorfs initialize" "" "\fLdorfs\fR \(em start and stop RFS automatically" ""
.IX RFS "start and stop automatically" "\fBRFS\fR" "start and stop automatically"
.IX start "RFS automatically" "start" "\s-1RFS\s0 automatically"
.IX stop "RFS automatically" stop "\s-1RFS\s0 automatically"
.IX initialize RFS initialize \s-1RFS\s0
.LP
.B dorfs 
sets up necessary environment to run 
Remote File Sharing
(\s-1RFS\s0).
You can also use it to start or stop
.SM RFS 
automatically,
after its environment is initialized.
The environment only needs to be set up once and
.BR /usr/nserve/rfmaster
must exist before the environment is initialized.
Descriptions of 
.BR /usr/nserve/rfmaster 
are in
.BR rfmaster (5).
You must be the super-user to run this command.
.SH USAGE
.SS Subcommands
.HP
.B init
.IR "\ domain\ \ netspec" [\| address \|]  	
.br
.I domain 
is the name of the
.SM RFS
domain.
.I netspec 
is the name of a device file in the
.B /dev 
directory which represents the 
streams-based transport provider on which
.SM RFS
will run.
Currently,
.B tcp
is the only accepted value for this field.
.I address
is the optional
.B tcp
port number on which the listener will listen.
If unspecified, it defaults to
.BR 0x1450 .
This subcommand only needs to be run once to
initialize the environment.
You do not need to rerun
.B dorfs
with the
.B init  
argument, unless you want to change
.IR  netspec .
.BR /usr/nserve/rfmaster 
must exists before you run
this command to initialize the environment.
To reinitialize the environment, you need to remove 
.BR /usr/nserve/\fIdomain ,
.BR /usr/nserve/\fInetspec ,
.BI /var/net/nls/ netspec / address
and 
.BI /var/net/nls/ netspec /dbf
beforehand.
.HP 
.B start
[
.B \-v
]
.br
Start
.SM RFS
automatically.
.\"If the 
.\".I netspec 
.\"is 
.\".BR tcp , 
.\"it starts 
.\".BR listener ,
.\".BR in.named , 
.\".B rfudaemon 
.\"and 
.\".BR rfstart . 
It also automatically 
advertises resources that are stored in 
.B /etc/rstab 
and mounts 
.SM RFS 
resources that are stored in 
.BR /etc/fstab . 
.RS
.RS
.TP
.B \-v
Verify clients on mounts (see
.RB ` "rfstart \-v" ').
.RE
.RE
.TP
.B stop	
Takes down 
.SM RFS 
by forced unmounting of all advertised resources, umounting 
all remotely mounted resources, executing 
.BR rfstop ,
and stopping
.BR listener .
.SH FILES
.PD 0
.TP 25
.B /etc/advtab
.TP
.B /etc/rstab
.TP
.B /var/net/nls/tcp/addr
.TP
.B /var/net/nls/tcp/dbf
.TP
.B /usr/nserve/domain
.TP
.B /usr/nserve/netspec
.TP
.B /usr/nserve/rfmaster
.PD
.SH "SEE ALSO"
.BR rfmaster (5),
.BR dname (8),
.BR fumount (8),
.BR mount (8),
.BR nlsadmin (8),
.BR rfstart (8),
.BR rfstop (8)


usr.etc/rfs/startup/dorfs.sh
#! /bin/sh
#
# 	Start/stop RFS automatically 
#
#
#       %Z%  %M% %I% %E%
#
#       Copyright (c) 1987 by Sun Microsystems, Inc.
#

HOME=/; export HOME
PATH=/bin:/usr/bin:/etc:/usr/etc:/usr/ucb

CMDNAME=$0
USAGE="usage: $0 <start [-v] | stop | init <domain> <netspec> [addr]>"

#
# Verify arguments
#
if [ $#  -lt 1 ]; then
	echo $USAGE
	exit 1
fi
OP=$1
case "$OP" in
"start" )
	if [ $# -lt 1 -o $# -gt 2 ]; then
        	echo "${CMDNAME}: incorrect number of arguments."
		echo $USAGE
        	exit 1
	else
		if [ -f /usr/nserve/rfmaster -a -f /usr/nserve/domain -a \
			-f /usr/nserve/netspec ]; then
			if [ $# -eq 2 -a "$2" = "-v" ]; then
				VFLAG="-v"
			else
				VFLAG=""
			fi
			NETSPEC=`cat /usr/nserve/netspec`
		fi
	fi
	break ;;
"stop" )
        if [ $# -ne 1 ]; then
                echo "${CMDNAME}: incorrect number of arguments."
                echo $USAGE
                exit 1    
        else
		if [ -f /usr/nserve/rfmaster -a -f /usr/nserve/domain -a \
                        -f /usr/nserve/netspec ]; then
                	NETSPEC=`cat /usr/nserve/netspec`
		fi
        fi
        break ;;
"init" )
	if [ $# = 3 ] ; then
		DOMAIN=$2
		NETSPEC=$3
		ADDR=5200
	elif [ $# = 4 ] ; then
		DOMAIN=$2
		NETSPEC=$3
		ADDR=$4
	else
        	echo "${CMDNAME}: incorrect number of arguments."
        	echo $USAGE 
        	exit 1
	fi 
	break ;;
* )
	echo "${CMDNAME}: incorrect option specified."
	echo $USAGE
	exit 1 ;;
esac

case "$OP" in
"init" )
        #
        # Initialize domain and netspec
        #
        rm -f /usr/nserve/domain /usr/nserve/netspec
        dname -D $DOMAIN 
        if [ $? != 0 ]; then
                echo
                echo "${CMDNAME}: can not start dname."
		echo "${CMDNAME}: RFS not started."
                exit 1
        fi
        dname -N $NETSPEC
        if [ $? != 0 ]; then 
                echo 
                echo "${CMDNAME}: can not start dname." 
                echo "${CMDNAME}: RFS not started." 
                exit 1 
        fi

        #
        # Initialize netspec
        #
        nlsadmin -i $NETSPEC
	if [ $? != 0 ]; then 
                echo 
                echo "${CMDNAME}: can not initialize $NETSPEC."
                echo "${CMDNAME}: RFS not started."
                exit 1 
        fi
	if [ $NETSPEC = "npack" ] ; then
        	nlsadmin -l \\x00000001${ADDR}0000 $NETSPEC
		if [ $? != 0 ]; then  
                	echo  
 ad              	echo "${CMDNAME}: can not execute nlsadmin." 
                	echo "${CMDNAME}: RFS not started." 
                	exit 1  
        	fi
        	nlsadmin -t \\x00000002${ADDR}0000 $NETSPEC
		if [ $? != 0 ]; then   
                	echo   
                	echo "${CMDNAME}: can not execute nlsadmin."  
                	echo "${CMDNAME}: RFS not started."  
                	exit 1   
        	fi
	fi 
	if [ $NETSPEC = "tcp" ] ; then
        	nlsadmin -l `hostrfs \`hostname\` $ADDR` $NETSPEC
		if [ $? != 0 ]; then  
                	echo  
                	echo "${CMDNAME}: can not execute nlsadmin." 
                	echo "${CMDNAME}: RFS not started." 
                	exit 1  
        	fi
        	nlsadmin -t `hostrfs \`hostname\` \`expr $ADDR + 1\`` $NETSPEC
		if [ $? != 0 ]; then   
                	echo   
                	echo "${CMDNAME}: can not execute nlsadmin."  
                	echo "${CMDNAME}: RFS not started."  
                	exit 1   
        	fi
	fi 
	if [ -f /var/net/nls/${NETSPEC}/dbf ]; then
        	grep rfsetup /var/net/nls/${NETSPEC}/dbf >/dev/null
        	if [ $? = 1 ]; then
                	nlsadmin -a 105 -c /usr/net/servers/rfs/rfsetup -y "RFS Server" $NETSPEC
			if [ $? != 0 ]; then  
                		echo  
                		echo "${CMDNAME}: can not execute nlsadmin."
                		echo "${CMDNAME}: RFS not started." 
                		exit 1   
        		fi
		fi
	else
		echo "${CMDNAME}: /usr/net/nls/npack/dbf does not exist."	
		exit 1
        fi ;;
"start" )
	if [ -f /usr/nserve/rfmaster -a -f /usr/nserve/domain -a -f /usr/nserve/netspec ]; then
		#
		# Start npackd
		#
		if [ $NETSPEC = "npack" ] ; then
			npackd 2>/dev/null &
			if [ $? != 0 ]; then
				ps ax | grep npackd > /tmp/TMP
                		ID=`cat /tmp/TMP | awk '$5 == "npackd" { print $1 }'`
				if [ "$ID" = "" ]; then
                			echo
                			echo "${CMDNAME}: can not start npackd."
					exit 1
				fi
        		fi
		fi

		#
		# Start listener
		#
		rm -f /usr/net/nls/${NETSPEC}/lock
		nlsadmin -s $NETSPEC &
		if [ $? != 0 ]; then
			ps ax | grep listen > /tmp/TMP
                	ID=`cat /tmp/TMP | awk '$5 == "listen" { print $1 }'`
                	if [ "$ID" = "" ]; then
                		echo
                		echo "${CMDNAME}: can not start listener."
				# 
        			# Stop npackd
        			#
				if [ $NETSPEC = "npack" ] ; then
        				ps ax | grep npackd > /tmp/TMP
        				ID=`cat /tmp/TMP | awk '$5 == "npackd" { print $1 }'`
        				kill -9 $ID
        				rm -f /tmp/TMP
                			exit 1
				fi 
			fi
        	fi

		#
		# Start RFS : nserve and fudaemon
		#
		rfstart $VFLAG
		if [ $? != 0 ]; then
                	echo
                	echo "${CMDNAME}: can not start rfstart."
			#  
                	# Stop npackd 
                	#
			if [ $NETSPEC = "npack" ] ; then 
                		ps ax | grep npackd > /tmp/TMP
                		ID=`cat /tmp/TMP | awk '$5 == "npackd" { print $1 }'` 
                		kill -9 $ID
                		rm -f /tmp/TMP
			fi
			#
        		# Stop listener
        		#
        		nlsadmin -k $NETSPEC
                	exit 1 
        	fi

		#
		# Advertise resources in /etc/rstab
		#
		if [ -f /etc/rstab ]; then
			/etc/rstab
		fi
		#
		# Mount resources
		#
		umount -vat rfs > /dev/null 2>&1
		mount -vat rfs -o bg

		echo
		echo "RFS started." 
	else
		echo
                echo "RFS is not initialized and RFS is not started."
	fi
	break ;;
"stop" )
	if [ -f /usr/nserve/rfmaster -a -f /usr/nserve/domain -a \
                        -f /usr/nserve/netspec ]; then
		#
		# Force umount all advertised resources
		#
		if [ -f /etc/advtab ]; then
			for i in `awk '{ print $1 }' < /etc/advtab`; do
				fumount $i
			done
		fi
		#
		# Umount all remotely mounted resources
		#
		mount > /tmp/FSTAB
		for i in `awk '$5 == "rfs" { print $3 }' < /tmp/FSTAB`; do
                	umount -d $i
        	done
		rm -f /tmp/FSTAB
		#
		# Stop RFS : nserve and fudaemon
		#
		rfstop
		if [ $? != 0 ]; then
                	echo
                	echo "${CMDNAME}: can not start rfstop."
                	exit 1
        	fi

		#
		# Stop npackd
		#
		if [ $NETSPEC = "npack" ] ; then
			ps ax | grep npackd > /tmp/TMP
			ID=`cat /tmp/TMP | awk '$5 == "npackd" { print $1 }'`
			kill -9 $ID
			rm -f /tmp/TMP
		fi

		#
		# Stop listener
		#
		nlsadmin -k $NETSPEC
		if [ $? != 0 ]; then
                	echo
                	echo "${CMDNAME}: RFS not stopped."  
                	exit 1
        	fi

		echo
		echo "RFS stopped." 
	else
		echo
                echo "RFS is not initialized and RFS is not running."
	fi
	break ;;
esac
exit 0
